runtime/hashing/common/src/aws/smithy/kotlin/runtime/hashing/Crc32c.kt

```diff
@@ -0,0 +1,676 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.hashing
+
+import aws.smithy.kotlin.runtime.util.InternalApi
+import kotlin.experimental.and
+import kotlin.experimental.xor
+
+@InternalApi
+public abstract class Crc32cBase : HashFunction {
+    override val blockSizeBytes: Int = 4
+    override val digestSizeBytes: Int = 4
+
+    public abstract fun digestValue(): UInt
+}
+
+/**
+ * CRC32C checksum. Note: [digest] will return the bytes (big endian) of the CRC32C integer value. Access [digestValue]
+ * directly to avoid doing the integer conversion yourself.
+ */
+@InternalApi
+public expect class Crc32c() : Crc32cBase
+
+/**
+ * Compute the CRC32C hash of the current [ByteArray]
+ */
+@InternalApi
+public fun ByteArray.crc32c(): UInt = Crc32c().apply { update(this@crc32c) }.digestValue()
+
+public actual class Crc32c : Crc32cBase() {
```

Comment:
Either
(1) remove `expect`/`actual` all together (only common should have `expect`, `actual` is used by each platform to provide a real/actual implementation of the expected thing) and simplify the type hierarchy here
OR
(2) move `CRC32CImpl` to it's own file making it `internal` and move this `actual` to `Crc32cJVM.kt` in the JVM sourceset

Reply:
I think I misinterpreted your previous comment about keeping expect / actual in case we want to use CRT bindings for Kotlin Native. For now I've removed the `expect` / `actual` altogether